### PR TITLE
Explicitly shutdown bazel after the target finishes

### LINKED ISSUE #87441: [Flaky Test] Stage
**Which jobs are flaking**:
pull-kubernetes-e2e-gce
pull-kubernetes-e2e-gce-rbe

**Which test(s) are flaking**:
Stage

**Testgrid link**:
https://testgrid.k8s.io/presubmits-kubernetes-nonblocking#pull-kubernetes-e2e-gce-rbe&include-filter-by-regex=Overall%7CBuild%7CStage
https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-e2e-gce&include-filter-by-regex=Overall%7CBuild%7CStage

**Reason for failure**:
This isn't technically a test that's failing, it's part of the build->stage->up->test cycle used by our e2e jobs.

There are two main clusters of failures that I can tell.  One is timeouts.

The other looks like:
```
- Hashing and copying public release artifacts to
gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-gce/v1.18.0-alpha.2.14+1b7738cc4ebdbe:
$TEST_TMPDIR defined: output root default is '/bazel-scratch/.cache/bazel' and max_idle_secs default is '15'.
Server terminated abruptly (error code: 14, error message: '', log file: '/bazel-scratch/.cache/bazel/_bazel_prow/cae228f2a89ef5ee47c2085e441a3561/server/jvm.out')
Signal ERR caught!
Traceback (line function script):
208 main /home/prow/go/src/k8s.io/release/push-build.sh
Exiting...
```
Example -gce jobs:
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/85861/pull-kubernetes-e2e-gce/1219647945268269056 (#85861)
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/85846/pull-kubernetes-e2e-gce/1219646811342376960 (#85846)
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/87422/pull-kubernetes-e2e-gce/1219612336432615426 (#87422)
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/87372/pull-kubernetes-e2e-gce/1219266675761745921 (#87372)

Example -gce-rbe jobs:
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/87435/pull-kubernetes-e2e-gce-rbe/1219718868767870977
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/87266/pull-kubernetes-e2e-gce-rbe/1219655997224652803
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/86408/pull-kubernetes-e2e-gce-rbe/1219620136827228160

(source gist https://gist.github.com/spiffxp/ed54d006e630b8fd8126ddfe67d9dc1e)

**Anything else we need to know**:
- Triage link for the job that actually reports to PRs: https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&job=gce%24&test=Stage
- [The line where the crash is happening]( https://github.com/kubernetes/release/blob/f59f6ba61ac5f225cde91ff0fc179a0a6f762841/lib/releaselib.sh#L1148) ends up running `bazel run //:push-build $gcs_stage $gcs_destination`
- Some googling for "bazel error code 14" leads me to believe that bazel (or its jvm) is getting killed due to OOM.  [This job requests 6Gi of memory](https://github.com/kubernetes/test-infra/blob/b2471685eed6a7d063d7e1e19032282bb33679db/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L59-L61) which is a value that's been cargo culted around as far as I can tell.  I was unable to find the decision that led to this value.  I'm wondering if a bump to 8Gi of memory would help reduce occurrences of this flake.
- I'm referencing the -rbe jobs to point out that this happens even if we use RBE, but they're not blocking

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
/kind cleanup

variation of @spiffxp 's https://github.com/kubernetes/test-infra/pull/16071/

**What this PR does / why we need it**:
Try to see if this helps with the bazel OOMs / interrupts / timeouts. Please see additional info in https://github.com/kubernetes/test-infra/pull/16071/#issue-369237394

**Which issue(s) this PR fixes**:

Fixes #87441

**Special notes for your reviewer**:
- `Server terminated abruptly` as seen [here](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/87835/pull-kubernetes-e2e-gce/1225140837730488320/build-log.txt)
- `process.go:199: Interrupt after 1h20m0s timeout` as seen [here](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/87835/pull-kubernetes-e2e-gce/1225164126720888833/build-log.txt) 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
PASS 1 (contd):
- containerd and typecheck failures not related to bazel (pod timeout?)

**Comment by dims** [comment]
cross `Build timed out` ...

/test pull-kubernetes-cross

**Comment by dims** [comment]
PASS2 : gce job had 4 flakes, No OOM/interrupt or timeout ( https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/87866/pull-kubernetes-e2e-gce/1225249424582643712/ )

/test pull-kubernetes-e2e-gce

**Comment by dims** [comment]
/assign @BenTheElder @cblecker 

Ben, Christoph, i'd prefer this PR over kubernetes/test-infra#16071 as it is easier to test/revert and does not get in the way of anything else. PTAL.

File: build/root/Makefile [hunk 1]
```diff
@@ -623,4 +623,5 @@ bazel-release:
 else
 bazel-release:
 	bazel build //build/release-tars
+	bazel shutdown
```
Comment (liggitt) [inline build/root/Makefile]: should we do the same thing after `bazel build -- //... -//vendor/...` in the `bazel-build:` target, or is there a reason just to do it here?

Comment (dims) [reply build/root/Makefile]: @liggitt i don't see evidence of any CI jobs that fail during that step. So i don't want to add it anywhere else. We have clear evidence in the logs that the next bazel call after `build //build/release-tars` fails, so just adding it here.

Comment (liggitt) [reply build/root/Makefile]: ok

**Comment by cblecker** [comment]
I don't see the harm in this. I'd like @spiffxp to weigh in if he's already been working on this bazel stuff.

/lgtm
/approve
/hold

**Comment by dims** [comment]
@cblecker Aaron may not be available today from what i understand. It's a one liner and it's only for the release target and it's exactly the same as what aaron proposed but in a different spot. I'll baby sit any fallouts.

/hold cancel

**Comment by liggitt** [comment]
> Aaron may not be available today from what i understand. It's a one liner and it's only for the release target and it's exactly the same as what aaron proposed but in a different spot. I'll baby sit any fallouts.

That's correct. I'm in favor of proceeding as well. This is simple to revert if needed and this has 100% e2e build/stage success over 4 runs, which is significantly better than the average PR right now.

**Comment by cblecker** [comment]
Worst case scenario I can see is increased CI job time, and potential timeouts. I'm fine with proceeding.

File: build/root/Makefile (hunk 1 above)
Comment (BenTheElder) [reply build/root/Makefile]: we're going to start it again for push-build, this will cost us some time.

**Comment by BenTheElder** [comment]
We should do this, but be aware that the bazel server staying up keeps the repo analyzed, I'm pretty sure we have to rebuild that before push build now and that's actually kinda slow.

This sort of thing has me questioning the actual build speed gains we get for e2e targets. I'd like to get some hard data comparing ~~`make quick-release` or whatever the other kubetest jobs use versus this.

**Comment by BenTheElder** [comment]
My current anecdata is that over time it hasn't gotten as helpful, but we should get hard data :+)

**Comment by BenTheElder** [comment]
/lgtm
/approve

though also noting that for local users this is suboptimal behavior, that said I don't have significant evidence of local usage...

**Comment by liggitt** [comment]
stage looks way greener since this merged: https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-e2e-gce&include-filter-by-regex=%5E(Stage%7CTimeout)%24&width=5

**Pull request merged**